### ISSUE TITLE
Support for other online services

### ISSUE THREAD

**Issue by korpe1** [audacity/audacity#6332]
### Your idea

Hey, can we add support also for other online services than just audio.com? 

### Problem to be solved

.

**Comment by LWinterberg** [comment]
What exactly do you have in mind?

**Comment by korpe1** [comment]
> What exactly do you have in mind?

What do you mean? I think, my idea is pretty simple. Now Audacity supports only audio.com, but what if I want to use something else? Maybe just something sftp, network drive, vanilla Nextcloud or Nextcloud with plugin (where are thousands of them).

**Comment by LWinterberg** [comment]
Right. So, the cloud saving feature is a bit more complicated than simply a bunch of full projects backed up on an FTP server. Instead, saves are incremental, and converting an incremental save back into a full AUP3 would need server-side software. 

From the options you listed, a Nextcloud plugin might be viable and we'll certainly support community efforts to get such a plugin going. As for us, we have big plans to improve Audacity-related cloud saving features on Audio.com, since we have control over both ends, so that's where our current focus is.

**Comment by korpe1** [comment]
> Right. So, the cloud saving feature is a bit more complicated than simply a bunch of full projects backed up on an FTP server. Instead, saves are incremental, and converting an incremental save back into a full AUP3 would need server-side software.

I understand that bare FTP can't support all features we have with cloud.com, but part of them surely.

> From the options you listed, a Nextcloud plugin might be viable and we'll certainly support community efforts to get such a plugin going. As for us, we have big plans to improve Audacity-related cloud saving features on Audio.com, since we have control over both ends, so that's where our current focus is.

I main wish is that offer something stable API/standard for cloud support and then Nextcloud and others can build support for it. AFAIK by now it's hardcoded for audio.com url and also the UI offer only audio.com.

**Comment by korpe1** [comment]
> Instead, saves are incremental, and converting an incremental save back into a full AUP3 would need server-side software.

And I think there are many traditional software that support incremental savings e.g. rsync and git.

**Comment by korpe1** [comment]
> As for us, we have big plans to improve Audacity-related cloud saving features on Audio.com, since we have control over both ends, so that's where our current focus is.

This is understandable, but in this universal, open-source application I don't want to see deeply integration with only one proprietary cloud service. The open source way is in standards.

So, please always implement way to also other online services support the same features and keep the online service spesific parts of the source code in their own modules/plugins.
